src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -183,46 +182,55 @@ private void UpdateSize()
             if (extraField.Tag != TagConstant)
                 return false;
 
-            // this pattern needed because nested using blocks trigger CA2202
-            MemoryStream? ms = null;
-            try
-            {
-                ms = new MemoryStream(extraField.Data);
-                using (BinaryReader reader = new BinaryReader(ms))
-                {
-                    ms = null;
+            zip64Block._size = extraField.Size;
 
-                    zip64Block._size = extraField.Size;
+            using (MemoryStream ms = new MemoryStream(extraField.Data))
```

Comment:
extraField.Data is just a byte array, so instead of streams this could be BinaryPrimitives around a Span around the array. But then you have to add slicing.